File: src/main/java/sample/engine/canvasui/painter/PainterActionSimple.java
package sample.engine.canvasui.painter;

import javafx.scene.canvas.GraphicsContext;

public abstract class PainterActionSimple
        extends PainterAction {
    @Override
    public void paint(Painter painter) {
        doPaint(painter._left, painter._top, painter._gc);
    }
    public abstract void doPaint(int left, int top, GraphicsContext gc);
}

File: src/main/java/sample/engine/canvasui/painter/actions/PainterActionSimple.kt
package sample.engine.canvasui.painter.actions

import javafx.scene.canvas.GraphicsContext
import sample.engine.canvasui.painter.Painter

abstract class PainterActionSimple : PainterAction() {
    override fun paint(painter: Painter) {
        doPaint(painter._left, painter._top, painter._gc!!)
    }

    abstract fun doPaint(left: Int, top: Int, gc: GraphicsContext)
}